services/ndd_service.js

```diff
@@ -4,155 +4,106 @@
-          resolve(null);
-        } else {
-          res.setEncoding('utf8');
-          let buffer = '';
-          res.on('data', data => buffer += data);
-          res.on('end', _ => {
-            try {
-              resolve(JSON.parse(buffer));
-            } catch (e) {
-              resolve(null);
-            }
-          });
-        }
-      }).on('error', _ => resolve(null));
-    });
+  dispose() {
+    process.disconnect();
   }
 
-  async dispose() {
-    try {
-      for (const id of Array.from(this._running)) {
-        try {
-          process.kill(id, 'SIGKILL');
-        } catch (e) {
-        }
-      }
-      this._running.clear();
-      for (const ws of this._sockets.values())
-        ws.close();
-      for (const watcher of this._nddStoreWatchers)
-        watcher.close();
-      await removeFolder(this._nddStores[0]);
-      this._nddStores = [];
-      this._nddStoreWatchers = [];
-    } catch (e) {
-    } finally {
-      process.exit(0);
+  async _runSession(info, frontend, resumeTarget) {
+    const ws = new WebSocket(info.inspectorUrl);
+    const openPromise = new Promise(resolve => ws.once('open', () => resolve(DebugState.WS_OPEN)));
+    const errorPromise = new Promise(resolve => ws.once('error', () => resolve(DebugState.WS_ERROR)));
+    const closePromise = new Promise(resolve => ws.once('close', () => resolve(DebugState.WS_CLOSE)));
+    let state = await Promise.race([openPromise, errorPromise, closePromise, this._disconnectPromise]);
+    if (state === DebugState.WS_OPEN) {
+      const messageListener = messageString => {
+        const message = JSON.parse(messageString);
+        message.sessionId = info.id;
+        protocolDebug('<', message);
+        frontend.dispatchMessage(message);
+      };
+      ws.on('message', messageListener);
+      this._sockets.set(info.id, ws);
+      state = await Promise.race([frontend.detected(info), this._disconnectPromise]);
+      resumeTarget();
+      if (state !== DebugState.PROCESS_DISCONNECT)
+        state = await Promise.race([closePromise, errorPromise, this._disconnectPromise]);
+      ws.removeListener('message', messageListener);
+      this._sockets.delete(info.id);
+      if (state !== DebugState.PROCESS_DISCONNECT)
+        frontend.disconnected(info.id);
+      else
+        ws.send(CALL_EXIT_MESSAGE, () => ws.close());
+    } else {
+      resumeTarget();
     }
   }
 
-  debug(execPath, args, options) {
-    let nodePath = process.env.NODE_PATH || '';
-    if (nodePath)
-      nodePath += process.platform === 'win32' ? ';' : ':';
-    nodePath += path.join(__dirname, '..', 'lib', 'preload');
-    const env = {
-      NODE_OPTIONS: `--require ndb/preload.js`,
-      NODE_PATH: nodePath,
-      NDD_STORE: this._nddStores[0],
+  env() {
+    return {
+      NODE_OPTIONS: `--require ndb/preload.js --inspect=0`,
+      NODE_PATH: `${process.env.NODE_PATH || ''}${path.delimiter}${path.join(__dirname, '..', 'lib', 'preload')}`,
```

Comment:
`path.join(process.env.NODE_PATH || '', __dirname, '..', 'lib', 'preload')`

Reply:
path.join works great for paths and adds path.sep as delimeter but it does not work for PATH environment where path.delimeter should be used.